package/MDAnalysis/analysis/msd.py

```diff
@@ -263,7 +285,17 @@
 del Doi
 
 
+
 class EinsteinMSD(AnalysisBase):
+
+    @staticmethod
+    def is_linear(sequence): # This function is to check if the timesteps in the array is linear or not! If it's linear default MDA code will carry-out other wise the modified code carries!
+        if len(sequence) < 2:
+            return True
+        step = sequence[1] - sequence[0]
+        return all(sequence[i+1] - sequence[i] == step for i in range(len(sequence) - 1))
```

Comment:
So right now, it's failing the test.  My guess is that it's going down the wrong test path, because it's interpreting whatever data is being passed in by the test as not actually being linear (maybe something like 1.0000 gets interpreted at 0.999999).  A couple of options here:

1. Add a keyword like `nonlinear` that controls the logic, rather than determining the logic from the steps. 
2. Using something like assert_almost_equal so floating point errors don't cause the logic to give the wrong answer (i.e. that they are not equally spaced when they actually are).